registrations/serializers.py

```diff
@@ -159,10 +203,16 @@ def update(self, instance, validated_data):
         protected_data = _get_protected_data(
             validated_data, ["extra_info", "date_of_birth"]
         )
+        contact_person_data = validated_data.pop("contact_person", None) or {}
 
         super().update(instance, validated_data)
         self._update_or_create_protected_data(instance, **protected_data)
 
+        self._update_or_create_contact_person(instance, **contact_person_data)
+        contact_person = getattr(instance, "contact_person", None)
+        if instance.signup_group_id and contact_person:
+            contact_person.delete()
```

Comment:
A signup in a group shouldn't have a contact person of it's own. It's the group that should have a single shared contact person. This code was meant to remove a signup's own contact person if the signup was added to a group through an update, but yeah, the implementation wasn't very optimal.

I've changed it now so that `_update_or_create_contact_person` is called only if the signup doesn't belong to a group. Alternatively the contact person is deleted if the signup does belong to a group and has a contact person of its own.